noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -198,75 +195,41 @@ pub contract NFT {
-
-        // We create a setup payload with unpopulated/zero token id for 'to'
-        // TODO(#7775): Manually fetching the randomness here is not great. If we decide to include randomness in all
-        // notes we could just inject it in macros.
-
-        // Safety: We use the randomness to preserve the privacy of the note recipient by preventing brute-forcing,
-        // so a malicious sender could use non-random values to make the note less private. But they already know
-        // the full note pre-image anyway, and so the recipient already trusts them to not disclose this
-        // information. We can therefore assume that the sender will cooperate in the random value generation.
-        let note_randomness = unsafe { random() };
-        let note_setup_payload = NFTNote::setup_payload().new(to, note_randomness, to_note_slot);
-
-        let setup_log = note_setup_payload.encrypt_log(context, to, context.msg_sender());
-
-        // Using the x-coordinate as a hiding point slot is safe against someone else interfering with it because
-        // we have a guarantee that the public functions of the transaction are executed right after the private ones
-        // and for this reason the protocol guarantees that nobody can front-run us in consuming the hiding point.
-        // This guarantee would break if `finalize_transfer_to_private` was not called in the same transaction. This
-        // however is not the flow we are currently concerned with. To support the multi-transaction flow we could
-        // introduce a `from` function argument, hash the x-coordinate with it and then repeat the hashing in
-        // `finalize_transfer_to_private`.
-        //
-        // We can also be sure that the `hiding_point_slot` will not overwrite any other value in the storage because
-        // in our state variables we derive slots using a different hash function from multi scalar multiplication
-        // (MSM).
-        let hiding_point_slot = note_setup_payload.hiding_point.x;
-
-        // We don't need to perform a check that the value overwritten by `_store_point_in_transient_storage_unsafe`
-        // is zero because the slot is the x-coordinate of the hiding point and hence we could only overwrite
-        // the value in the slot with the same value. This makes usage of the `unsafe` method safe.
-        NFT::at(context.this_address())
-            ._store_payload_in_transient_storage_unsafe(
-                hiding_point_slot,
-                note_setup_payload.hiding_point,
-                setup_log,
-            )
-            .enqueue(context);
-
-        hiding_point_slot
+    ) -> PartialNFTNote {
+        // We create a partial note with unpopulated/zero token id for 'to'
+        let partial_note = NFTNote::partial(
+            to,
+            storage.private_nfts.at(to).storage_slot,
+            context,
+            to,
+            context.msg_sender(),
+        );
```

Comment:
(Not for this PR): It'd be cool if this could be created via the state variable. Like `private_nfts.at(to).new_partial(...)`